Initial 'GDP 11 HMI' integration

tom--pollard opened the pull request:
[GDP-409] New HMI integration into GDP Master

This PR includes the initial work to switch the GDP to the new HMI. It is also sent so others can experiment/test the current WIP.

tom--pollard committed:
Initial gdp-11-hmi recipe inclusion

Basic functionality can be tested with:

systemctl --user stop gdp-hmi-background  && systemctl --user stop gdp-hmi-panel &&
systemctl --user stop gdp-hmi-launcher2 && systemctl --user stop gdp-hmi-controller

/opt/genivi-11-hmi/bin/genivi-11-hmi -platform wayland

tom--pollard committed:
gdp-new-hmi: Add application manifests

Currently adds the existing GDP app manifest so the launcher app drawer
can be populated

tom--pollard committed:
gdp-new-hmi: Initial removal of old hmi, add new hmi to systemd

New hmi to be launched as a default target user service. The service
file will need tweaking after further testing

tom--pollard committed:
Move relevant metadata from launcher2 recipe into a seperate temp recipe

gdp-hmi-launcher2 contained steps to populate app metadata such as
icons. This needed to be exracted into a standalone recipe as a
temporary measure

tom--pollard committed:
Removed old hmi recipes & related files

Sweeping removal of the old hmi recipes and associated patches,
scripts and services. Also the removal of hmi-panel dependency from
new meta-data recipe.

tom--pollard committed:
gdp-new-hmi: Added patch to support compilation for rcar gen2

cxx11 error when compiling for rcar gen2 targets. This 'fix'
should land in the upstream repo if needed.

tom--pollard committed:
Removed old shared hmi.inc, updated qml example recipe to handle this

A .inc file was used across the old hmi recipes to inherit the same
upstream source. This have now been moved directly into the qml example
& metadata recipes for the time being.

tom--pollard commented:
There is currently an issue with app resolution / location on the porter, which I believe is coming from incorrectly sourced/set ivi surfaces. This issue is reported in the jira ticket

RobertAJMarshall commented:
Builds and runs for me (on rasp pi 3)

tom--pollard committed:
gdp-new-hmi: Update to latest commit

d0ca9d0 - Updated launcher to correctly launch fmradio, hvac, browser and connected home
9595108 - Modifed home screen to make the browser button launch the GDP browser
11460a8 - Changed default width and height to 1024 x 768 when screen size cannot be detected
8fed857 - Modified the instructions to mention -platform wayalnd and copying manifest files

tom--pollard committed:
HVAC: Add JLR HVAC application to new hmi launcher

This brings in the current tip of the HVAC application into the GDP,
with the package added to the hmi package list.

chbae commented on meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/gdp-new-hmi.bb:
```diff
@@ -0,0 +1,22 @@
+    "
+
+do_install_append() {
+}
```
line 21 and 22 are not required.

tom--pollard replied:
l'm not sure what github has done here, but the do_install is definitely populated in the actual file

chbae replied:
This is empty and not required now. :)

RobertAJMarshall replied:
We've looked at the file and the do_install_append is not empty in the file (so we're confused by your comment!)

tom--pollard replied:
@chbae Ah you're looking at the initial commit of the file, please check the final commit :)

chbae commented on meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/gdp-new-hmi.bb:
```diff
@@ -0,0 +1,22 @@
+inherit autotools pkgconfig qmake5
+
+FILES_${PN} += "\
+    ${bindir}/* \
```
I thought that this line is not required

chbae commented on meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/gdp-new-hmi.bb:
```diff
@@ -0,0 +1,22 @@
+FILES_${PN} += "\
+    ${bindir}/* \
+    ${libdir}/systemd/user/* \
+    ${libdir}/* \
```
I thought that this line is not required

tom--pollard replied:
I can remove the ${bindir}/\* & ${libdir}/systemd/user/\* , the wildcard ${libdir}/\* is needed for the systemd installations and lib/qt5 runtimes created by the package. I will fix this up

tom--pollard replied:
Rebased and removed as stated

chbae commented on meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/gdp-new-hmi.bb:
```diff
@@ -0,0 +1,22 @@
+
+DEPENDS = "dbus-c++ systemd wayland-ivi-extension qtquick1 qtbase"
+
+RDEPENDS_${PN} += "qtbase qtsvg"
```
why runtime dependency?

tom--pollard replied:
The app needs it at runtime, I tested it without svg but it fails without it

chbae replied:
qtsvg is fine. why qtbase?.. qtbase is in depends. it's enough that I thought

chbae commented on meta-genivi-dev/recipes-dev-platform/packagegroups/packagegroup-gdp-hmi.bb:
```diff
@@ -22,4 +22,5 @@ RDEPENDS_${PN} += "\
     genivi-dev-platform-hmi \
     qml-example \
     gdp-hmi-mouse-test \
+    gdp-new-hmi \
```
name is good? new-hmi??

tom--pollard replied:
I've asked Genivi for a name and have not been given anything different, this can be changed obviously at a later date

chbae commented on meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/gdp-new-hmi.bb:
```diff
@@ -0,0 +1,22 @@
+SRC_URI = "git://github.com/GENIVI/hmi-layout-gdp.git;branch=ics-launcher-gdp"
```
Why don't you add "SUMMARY"?

tom--pollard replied:
Sure a summary can be added once one is provided

RobertAJMarshall committed:
Add other ICS applications to the HMI

[GDP-412] add ICS applications to the home screen
and add them to the hmi package list
